File: backend/src/main/java/com/example/nuevo_core/loan/AmortizationTable.java
package com.example.nuevo_core.loan;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class AmortizationTable {
    private long id;
    private long loanId;
    private List<AmortizationTableItem> item;


}

@Data
@Builder
class AmortizationTableItem {
    private UUID reference;
    private int paymentNumber;
    private BigDecimal cuota;
    private BigDecimal capital;
    private BigDecimal interes;
    private BigDecimal saldo;
    private LocalDate paymentDate;
    private boolean isPaid;
    private LocalDate paidDate;
}

File: frontend/core-frontend/src/models/AmortizationTable.ts
export interface AmortizationTable {
  loanNumber: number;
  items: AmortizationTableItem[];
}

export interface AmortizationTableItem {
  id: number;
  installmentNumber: number;
  cuota: number;
  capital: number;
  interes: number;
  charges: number;
  saldo: number;
  paymentDate: Date;
  paidDate: Date | null;
  paid: boolean; // puede ser null si no está pagado
}
